testing/integration/proxy_url_test.go

```diff
@@ -341,6 +359,229 @@ func TestProxyURL(t *testing.T) {
+						if strings.Contains(
+							r,
+							fleetservertest.NewPathCheckin(mockFleet.policyData.AgentID)) {
+							return true
+						}
+					}
+
+					return false
+				}, 5*time.Minute, 5*time.Second) {
+					t.Errorf("did not find requests to the proxy defined in the policy")
+				}
+
+				inspectOutput, err := fixture.Exec(ctx, []string{"inspect"})
+				assert.NoError(t, err, "error running elastic-agent inspect")
+				t.Logf("elastic-agent inspect output:\n%s\n", string(inspectOutput))
+			},
+		},
+		{
+			name: "NoEnrollProxy-mTLSProxyInThePolicy",
+			args: args{
+				// no proxy at enroll
+				enrollProxyName: "",
+			},
+			setup: func(t *testing.T, mockFleet *mockFleetComponents) (proxies map[string]*proxytest.Proxy) {
+
+				// use os.MkdirTemp since we are installing agent unprivileged and t.TempDir() does not guarantee that the elastic-agent user has access
+				baseDir := ""
+				if runtime.GOOS == "windows" {
+					baseDir = "C:\\"
+				}
+				tmpDir, err := os.MkdirTemp(baseDir, "mTLSProxyInThePolicy*")
+				require.NoError(t, err, "error creating temp dir for TLS files")
+
+				t.Cleanup(func() {
+					cleanupErr := os.RemoveAll(tmpDir)
+					assert.NoErrorf(t, cleanupErr, "error cleaning up directory %q", tmpDir)
+				})
+
+				// fix permissions on temp dir
+				err = os.Chmod(tmpDir, 0o755&os.ModePerm)
+				require.NoError(t, err, "error setting temporary dir %q as world-readable", tmpDir)
+
+				caKey, caCert, pair, err := certutil.NewRootCA()
+				require.NoError(t, err, "failed creating root CA")
+
+				caCertFile := filepath.Join(tmpDir, "ca.cert")
+				err = os.WriteFile(caCertFile, pair.Cert, 0o644&os.ModePerm)
+				require.NoError(t, err, "failed writing CA cert file %q", caCertFile)
+
+				caCertPool := x509.NewCertPool()
+				caCertPool.AddCert(caCert)
+
+				proxyCert, _, err := certutil.GenerateChildCert("localhost", []net.IP{net.IPv6loopback, net.IPv6zero, net.ParseIP("127.0.0.1")}, caKey, caCert)
+
+				// Create a fake proxy with mTLS config to be used in fleet policy
+				proxyFleetPolicy := proxytest.New(t,
+					proxytest.WithRequestLog("proxy-fleet-policy", t.Logf),
+					proxytest.WithVerboseLog(),
+					proxytest.WithServerTLSConfig(&tls.Config{
+						RootCAs: caCertPool,
+						Certificates: []tls.Certificate{
+							{
+								Certificate: proxyCert.Certificate,
+								PrivateKey:  proxyCert.PrivateKey,
+								Leaf:        proxyCert.Leaf,
+							},
+						},
+						// require client auth with a trusted Cert
+						ClientAuth: tls.RequireAndVerifyClientCert,
+						ClientCAs:  caCertPool,
+					}))
+				err = proxyFleetPolicy.StartTLS()
+				require.NoError(t, err, "error starting TLS-enabled proxy")
+				t.Cleanup(proxyFleetPolicy.Close)
+
+				// generate a certificate for elastic-agent from the same CA as the proxy
+				_, agentPair, err := certutil.GenerateChildCert("localhost", []net.IP{net.IPv6loopback, net.IPv6zero, net.ParseIP("127.0.0.1")}, caKey, caCert)
```

Comment:
[Suggestion]
I like to have different CAs for the client and server/proxy just for the sake of testing it as thorough as possible.

Reply:
Done in 78edd2e74371d82c4f6eb235908caa523e3244a4